src/core/control/tools/ArrowHandler.cpp

```diff
@@ -19,15 +19,38 @@ ArrowHandler::~ArrowHandler() = default;
 auto ArrowHandler::createShape(bool isAltDown, bool isShiftDown, bool isControlDown)
         -> std::pair<std::vector<Point>, Range> {
     Point c = snappingHandler.snap(this->currPoint, this->startPoint, isAltDown);
+    double lineLength = sqrt(pow(c.x - this->startPoint.x, 2) + pow(c.y - this->startPoint.y, 2));
+    double thickness = control->getToolHandler()->getThickness();
+    double slimness = lineLength / thickness;
 
     // We've now computed the line points for the arrow
-    // so we just have to build the head
-
-    // set up the size of the arrowhead to be 7x the thickness of the line
-    double arrowDist = control->getToolHandler()->getThickness() * 7.0;
+    // so we just have to build the head:
+    // arrowDist is the distance between the line's and the arrow's tips
+    // delta is the angle between each arrow leg and the line
 
     // an appropriate delta is Pi/3 radians for an arrow shape
     double delta = M_PI / 6.0;
+    // set up the size of the arrowhead to be 7x the thickness of the line
+    double arrowDist = thickness * 7;
+    if (slimness >= 7 / 0.4) {
+        // arrow is not too long compared to the line length
+    } else if (slimness >= 1.6 / 0.4) {
+        // arrow is not too short compared to the thickness
+        arrowDist = lineLength * 0.4;
+    } else if (slimness >= 1.6 / (doubleEnded ? 0.5 : 0.8)) {
+        // arrow is not too thick compared to the line length
+        arrowDist = thickness * 1.6;
+        // help visibility by widening the angle
+        delta = (1 + (slimness - 1.6 / 0.4) / (1.6 / (doubleEnded ? 0.5 : 0.8) - 1.6 / 0.4)) *  M_PI / 6.0;
+	// which allows to shorten the tips and keep the horizonzal distance
+	arrowDist *= sin(M_PI / 6.0) / sin(delta);
```

Comment:
```suggestion
        // which allows to shorten the tips and keep the horizonzal distance
        arrowDist *= sin(M_PI / 6.0) / sin(delta);
```
Something's wrong with our clang-format test unit... It should have caught this.
Could you apply clang-format to your changes?